src/main/java/org/example/daos/FileImportDao.java

```diff
@@ -0,0 +1,228 @@
+package org.example.daos;
+import javax.xml.transform.Result;
+import java.io.BufferedReader;
+import java.io.FileInputStream;
+import org.example.enums.Location;
+
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.sql.*;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+
+public class FileImportDao {
+
+    public int importRoles(final List<String> listOfLines) throws SQLException {
```

Comment:
This should be in a transaction as if one fails, duplicates will be accidentally added.